cdap-watchdog/src/main/java/co/cask/cdap/logging/write/LogCleanup.java

```diff
@@ -109,13 +121,127 @@ public Void call() throws Exception {
     }
   }
 
+  private void cleanFilesWithoutMeta(final long tillTime, final Map<String, NamespaceId> namespacedLogBaseDirMap,
+                                     final SetMultimap<String, Location> parentDirs) throws Exception {
+    LOG.info("Starting deletion of log files older than {} without metadata", tillTime);
+    List<NamespaceMeta> namespaceMetaList = namespaceQueryAdmin.list();
```

Comment:
does this operation run as the impersonated user? If so, that's the right thing to do? you shouldn't be able to clean up logs that you don't have access to anyway, right?